### PR TITLE
Move `Embed *` phases to before Bazel Integration script to fix Xcode 15 cyclic dependency error

### LINKED ISSUE #2514: Bug: Xcode 15 Beta 7 issue with Cycle inside watchOSApp
### Description

Logs:

```
error: Cycle inside watchOSApp (Simulator) (Debug) (53b1a); building could produce unreliable results. This usually can be resolved by moving the shell script phase 'Copy Bazel Outputs / Generate Bazel Dependencies (Index Build)' so that it runs before the build phase that depends on its outputs.
Cycle details:
→ Target 'watchOSApp (Simulator) (Debug) (53b1a)' has process command with output '/.../Xcode/DerivedData/Integration-cgsezinvjxaqevevgjhtbasohpah/Build/Products/Debug-watchsimulator/bazel-out/applebin_watchos-watchos_arm64-dbg-ST-b4b2917363cd/bin/watchOSApp/watchOSApp.app/Info.plist'
○ Target 'watchOSApp (Simulator) (Debug) (53b1a)' has copy command from '/.../Xcode/DerivedData/Integration-cgsezinvjxaqevevgjhtbasohpah/Build/Products/Debug-watchsimulator/bazel-out/applebin_watchos-watchos_arm64-dbg-ST-6464785a998a/bin/watchOSAppExtension/watchOSAppExtension.appex' to '/.../Xcode/DerivedData/Integration-cgsezinvjxaqevevgjhtbasohpah/Build/Products/Debug-watchsimulator/bazel-out/applebin_watchos-watchos_arm64-dbg-ST-b4b2917363cd/bin/watchOSApp/watchOSApp.app/PlugIns/watchOSAppExtension.appex'
○ That command depends on command in Target 'watchOSApp (Simulator) (Debug) (53b1a)': script phase “Copy Bazel Outputs / Generate Bazel Dependencies (Index Build)”


Raw dependency cycle trace:

target:  ->

node: <all> ->

command: <all> ->

node: <target-watchOSApp (Simulator) (Debug) (53b1a)-bd8833dedbef034af8134cf809cb7f19388f4690d73f0aa6a5ace58775614628--end> ->

command: P0:target-watchOSApp (Simulator) (Debug) (53b1a)-bd8833dedbef034af8134cf809cb7f19388f4690d73f0aa6a5ace58775614628-:Debug:Gate target-watchOSApp (Simulator) (Debug) (53b1a)-bd8833dedbef034af8134cf809cb7f19388f4690d73f0aa6a5ace58775614628--end ->

CYCLE POINT ->

node: /.../Xcode/DerivedData/Integration-cgsezinvjxaqevevgjhtbasohpah/Build/Products/Debug-watchsimulator/bazel-out/applebin_watchos-watchos_arm64-dbg-ST-b4b2917363cd/bin/watchOSApp/watchOSApp.app/Info.plist ->

command: P0:target-watchOSApp (Simulator) (Debug) (53b1a)-bd8833dedbef034af8134cf809cb7f19388f4690d73f0aa6a5ace58775614628-:Debug:ProcessInfoPlistFile /.../Xcode/DerivedData/Integration-cgsezinvjxaqevevgjhtbasohpah/Build/Products/Debug-watchsimulator/bazel-out/applebin_watchos-watchos_arm64-dbg-ST-b4b2917363cd/bin/watchOSApp/watchOSApp.app/Info.plist /Users/danao/Developer/rules_xcodeproj/examples/integration/bazel-output-base/rules_xcodeproj.noindex/build_output_base/execroot/_main/bazel-out/applebin_watchos-watchos_arm64-dbg-ST-b4b2917363cd/bin/watchOSApp/rules_xcodeproj/watchOSApp/Info.plist ->

node: /.../Xcode/DerivedData/Integration-cgsezinvjxaqevevgjhtbasohpah/Build/Products/Debug-watchsimulator/bazel-out/applebin_watchos-watchos_arm64-dbg-ST-b4b2917363cd/bin/watchOSApp/watchOSApp.app/PlugIns/watchOSAppExtension.appex ->

command: P0:target-watchOSApp (Simulator) (Debug) (53b1a)-bd8833dedbef034af8134cf809cb7f19388f4690d73f0aa6a5ace58775614628-:Debug:Copy /.../Xcode/DerivedData/Integration-cgsezinvjxaqevevgjhtbasohpah/Build/Products/Debug-watchsimulator/bazel-out/applebin_watchos-watchos_arm64-dbg-ST-b4b2917363cd/bin/watchOSApp/watchOSApp.app/PlugIns/watchOSAppExtension.appex /.../Xcode/DerivedData/Integration-cgsezinvjxaqevevgjhtbasohpah/Build/Products/Debug-watchsimulator/bazel-out/applebin_watchos-watchos_arm64-dbg-ST-6464785a998a/bin/watchOSAppExtension/watchOSAppExtension.appex ->

node: <target-watchOSApp (Simulator) (Debug) (53b1a)-bd8833dedbef034af8134cf809cb7f19388f4690d73f0aa6a5ace58775614628--fused-phase0-copy-bazel-outputs---generate-bazel-dependencies--index-build-> ->

command: P0:::Gate target-watchOSApp (Simulator) (Debug) (53b1a)-bd8833dedbef034af8134cf809cb7f19388f4690d73f0aa6a5ace58775614628--fused-phase0-copy-bazel-outputs---generate-bazel-dependencies--index-build- ->

node: <execute-shell-script-bd8833dedbef034af8134cf809cb7f1957b691bd88ffe68efd4da078c29ddb9f-target-watchOSApp (Simulator) (Debug) (53b1a)-bd8833dedbef034af8134cf809cb7f19388f4690d73f0aa6a5ace58775614628-> ->

command: P2:target-watchOSApp (Simulator) (Debug) (53b1a)-bd8833dedbef034af8134cf809cb7f19388f4690d73f0aa6a5ace58775614628-:Debug:PhaseScriptExecution Copy Bazel Outputs / Generate Bazel Dependencies (Index Build) /.../Xcode/DerivedData/Integration-cgsezinvjxaqevevgjhtbasohpah/Build/Intermediates.noindex/Integration.build/bazel-out/applebin_watchos-watchos_arm64-dbg-ST-b4b2917363cd/bin/watchOSApp/watchOSApp/Script-ED53254051A5B08FE6ED0FD9.sh ->

node: /.../Xcode/DerivedData/Integration-cgsezinvjxaqevevgjhtbasohpah/Build/Products/Debug-watchsimulator/bazel-out/applebin_watchos-watchos_arm64-dbg-ST-b4b2917363cd/bin/watchOSApp/watchOSApp.app/Info.plist
```

### Reproduction steps

1. Use Xcode 15 Beta 7
2. Run rules_xcodeproj example via bazel `bazel run //:xcodeproj-bazel-sim_arm64`
3. Select watchOSApp (Simulator) (Debug) target to build ()
4. Try to build

Potential fix: Reordering build phases

### Expected behavior

Target build successfully 

### rules_xcodeproj version

1.9.1

### Xcode version

Xcode 15, version15_0_0_15A5229h

### Bazel version

6.3.1-63bac8b

### rules_apple version

2.0.0

### rules_swift version

1.6.0

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by danaospo**
Fixes #2514.

### CODE REVIEW

**Comment by brentleyjones** [comment]
Can you produce a repro for the issue? I don’t think the new spot is correct, so I would like an example to debug.

**Comment by danaospo** [comment]
Hey, @brentleyjones!
We have validated rules_xcodeproj's given similar examples on Xcode 15, it works fine though. 
But we do have some complex setup on our project, where changing the order of Build Phases works fine, and we haven't come up with other solutions yet. 
The solution is not perfect and for now, marked it as draft. Will back soon as we found something better or if we try to reproduce it.

I can share that our logs of the issue are [similar to this example.](https://developer.apple.com/forums/thread/731825)

**Comment by danaospo** [comment]
Hey @brentleyjones!
We were able to reproduce the same issue on rules_xcodeproj examples.
Having `$(TARGET_BUILD_DIR)/$(INFOPLIST_PATH)` as Input Files in Run Script build phase makes Xcode fail to build a target. See the screenshot.
Removing it helps to build without reordering the build phases. Please try your own
Screenshot 2023-08-21 at 21 50 46

**Comment by danaospo** [comment]
@brentleyjones reopening the PR. Came up that it is better solution to reorder as Xcode 15 suggests

**Comment by danaospo** [comment]
> Does this work for BwX mode?

@brentleyjones yes, it does. But not with BwB

**Comment by danaospo** [comment]
> Does this work for BwX mode?

@brentleyjones Do you mean to check that changes on BwX mode?
The issue appears only on BwB

**Comment by brentleyjones** [comment]
Hmm. Since we generate the Info.plist in BwX mode as well, I expected an issue there. So I was wonder if the issue existed there, and that this change doesn't break anything there.

**Comment by brentleyjones** [comment]
Ohh, I see, this build phase doesn't exist in BwX mode, so the order isn't different there, right?

**Comment by danaospo** [comment]
so, when we build on BwX mode the  `inputPaths = ["$(TARGET_BUILD_DIR)/$(INFOPLIST_PATH)"] ` is ignored. When it is BwB it is included. And that's why it is causing the issue

**Comment by danaospo** [comment]
> Ohh, I see, this build phase doesn't exist in BwX mode, so the order isn't different there, right?
@brentleyjones 
yes

**Review by brentleyjones — APPROVED** [review]
Will need to update the fixtures with `./test/update_all_fixtures.sh` from the root of the repository.

File: examples/integration/test/fixtures/bwx.xcodeproj/project.pbxproj
Comment (brentleyjones) [inline examples/integration/test/fixtures/bwx.xcodeproj/project.pbxproj:9484]: And we confirmed this doesn't create an issue, like compiling starting later for apps in BwX mode?

**Review by brentleyjones — APPROVED** [review]

**Review by brentleyjones — APPROVED** [review]
Just waiting for some external BwX testing confirmation before merging.

File: examples/integration/test/fixtures/bwx.xcodeproj/project.pbxproj
Comment (danaospo) [reply examples/integration/test/fixtures/bwx.xcodeproj/project.pbxproj:9484]: Generated `Integration.xcodeproj` in both BwX & BwB mode and was able to build & run:
- iOS app ✅
- Watch app ✅
- macOS app ✅
Able to build extensions ✅

**Review by brentleyjones — APPROVED** [review]
(To run CI again.)

**Pull request merged**
